airflow/cli/commands/task_command.py

```diff
@@ -634,6 +635,12 @@ def task_test(args, dag: DAG | None = None) -> None:
                 session.delete(ti.dag_run)
 
 
+@provide_session
+def render_template_with_session(ti: TaskInstance, session=NEW_SESSION):
+    with set_current_task_instance_session(session=session):
+        ti.render_templates()
```

Comment:
Why does this need a separate function?

Reply:
To use decorator applired to a minimum scope it is needed. We do not need to keep session open after we already rendered the templates - i.e when we print the rendeder template values.

Reply:
I can also make it a nested function inside "task_render", but I am rather sceptical about adding functions inside functions. We could also potentially turn "provide_session" into a generic context manager but that would likely need some modifications to it and it would be introducing a new way of managing and allocating sessions - I would rather not introduce a new way of managing sessions but I would rely on how we do it elsewhere. 

I considered all those options and extracting the method the way I did, sounded like the most "straightforward" solution.

Reply:
There’s `create_session`, a decorator is not needed.

```python
with create_session() as session, set_current_task_instance_session(session=session):
    ti.render_templates()
```

Reply:
Right.

Reply:
Fixed.